app/src/main/java/live/hms/app2/ui/meeting/MeetingActivity.kt

```diff
@@ -82,5 +84,55 @@ class MeetingActivity : AppCompatActivity() {
     meetingViewModel.title.observe(this) {
       binding.containerToolbar.toolbar.setTitle(it)
     }
+    meetingViewModel.isRecording.observe(this) {
+      invalidateOptionsMenu()
+    }
+  }
+
+  override fun onPrepareOptionsMenu(menu: Menu?): Boolean {
+    super.onPrepareOptionsMenu(menu)
+
+    menu?.forEach { item ->
+      if (item.itemId != R.id.action_flip_camera && item.itemId != R.id.action_volume && item.itemId != R.id.action_participants) {
+        item.isVisible = false
+      }
+    }
+
+    menu?.findItem(R.id.action_record)?.apply {
+      when (meetingViewModel.isRecording.value) {
+        RecordingState.RECORDING -> {
+          // red
+          isVisible = true
+          this.icon.setTint(Color.parseColor("#e04848"))
+        }
+        RecordingState.NOT_RECORDING_OR_STREAMING -> {
+          isVisible = false
+        }
+        RecordingState.NOT_RECORDING_TRANSITION_IN_PROGRESS,
+        RecordingState.RECORDING_TRANSITIONING_TO_NOT_RECORDING -> {
+          // White
+          isVisible = true
+          // change the colour to transitioning
+          this.icon.setTint(Color.parseColor("#FFFFFF"))
+        }
+        RecordingState.STREAMING -> {
+          // Blue
+          isVisible = true
+          this.icon.setTint(Color.parseColor("#2832c2"))
+        }
+        RecordingState.STREAMING_AND_RECORDING -> {
+          // Orange
+          isVisible = true
+          this.icon.setTint(Color.parseColor("#FFC107"))
+        }
+        null -> TODO()
```

Comment:
we can remove tint incase of null

Reply:
Null is not a valid state so it's simply ignored, even in the current app if it actually hit that todo it would crash.

Reply:
yeah , so maybe we can remove the TODO() and add a Log or something